src/components/GlobalPreferences/CustomLabels/Search.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Button,
+  ButtonIcon,
+  GU,
+  IconCross,
+  IconSearch,
+  TextInput,
+  useTheme,
+  useLayout,
+  textStyle,
+} from '@aragon/ui'
+
+const SEARCH_COMPACT = Symbol('compact')
+const SEARCH_OPEN = Symbol('open')
+const EMPTY = ''
+
+const Search = React.memo(function Search({ onChange, value }) {
+  const { layoutName } = useLayout()
+  const compact = layoutName === 'small'
+  const theme = useTheme()
+
+  const [mode, setMode] = useState(compact ? SEARCH_COMPACT : SEARCH_OPEN)
```

Comment:
What would you think of using `SEARCH_CLOSED` / `SEARCH_OPENED` so that it describes a single thing?